src/components/timeout-dialog/timeout-dialog.browser.test.js

```diff
@@ -0,0 +1,408 @@
+        document.querySelector('.hmrc-timeout-dialog__message').textContent,
+        document.querySelector('#hmrc-timeout-message').textContent,
+      ];
+    });
+    expect(visibleMessage).toBe('For your security, we will sign you out in 0 seconds.');
+    expect(audibleMessage).toBe('For your security, we will sign you out in 20 seconds.');
+    completeSlowTimeoutRequest();
+    await page.waitForNavigation({ timeout: 500 });
+    await expect(page).toMatchTextContent('timeout page reached');
+  });
+
+  it('should display the time remaining as a whole number of seconds', async () => {
+    await renderTimeoutDialog(page, `
+      data-timeout="20"
+      data-countdown="10"
+      data-keep-alive-url="?keepalive"
+      data-sign-out-url="/timeout-reached"
+    `);
+    await clockTick(page, 13140);
+    await expect(page).toMatchTextContent('For your security, we will sign you out in 7 seconds.');
+    await clockTick(page, 2700);
+    await expect(page).toMatchTextContent('For your security, we will sign you out in 5 seconds.');
+    await clockTick(page, 160);
+    await expect(page).toMatchTextContent('For your security, we will sign you out in 4 seconds.');
+    await clockTickSeconds(page, 3);
+    await expect(page).toMatchTextContent('For your security, we will sign you out in 1 second.');
+  });
+
+  it('should incorporate a delay receiving session activity into new timeout countdown', async () => {
+    const session = await browser.createIncognitoBrowserContext();
+    const background = await session.newPage();
+    await renderTimeoutDialog(background, `
+      data-timeout="20"
+      data-countdown="10"
+      data-keep-alive-url="?keepalive"
+      data-sign-out-url="/timeout-reached"
+      data-synchronise-tabs="true"
+    `);
+    await clockTickSeconds(background, 15);
+    const foreground = await session.newPage();
+    await expect(background).toShowTimeoutDialog();
+    expect(await visualCountdownFrom(background))
+      .toBe('For your security, we will sign you out in 5 seconds.');
+    await useFakeTimers(foreground, { now: 10000 });
+    // by opening the new page with a clock initialized 5s in the past, we can fake that
+    // the session activity msg it broadcasts was received delayed by the bg tab, as if
+    // 5s passed since the fg page was opened, and test that the delay is incorporated in
+    // the calculation of time remaining in session by the bg page.
+    await render(foreground, withHmrcStylesAndScripts('so that we broadcast session activity'));
+    await background.bringToFront();
+    await expect(background).not.toShowTimeoutDialog();
+    await clockTickSeconds(background, 10);
+    await expect(background).toShowTimeoutDialog();
+    expect(await visualCountdownFrom(background))
+      .toBe('For your security, we will sign you out in 5 seconds.');
+    await session.close();
+  });
+
+  // TODO because currently it would actually log you out straight away, this is an edge case
```

Comment:
might be an edge case that in practice can't happen (I don't know if session activity could be delayed like this, so left this as a skipped test to show how it currently behaves)